### ISSUE TITLE
Update focus and title handling around loading state of detail pages

### ISSUE THREAD

**Issue by jbalboni** [department-of-veterans-affairs/va.gov-team#20725]
We need to update the new appointment detail pages to make sure that we're focusing on the header and loading indicators as the appointment is loaded and displayed on the detail pages. We also need to set an appropriate document.title.

AC
- When a detail page is open, focus should be put on the loading spinner and then the header when loading finishes
- If the detail page is loaded without a loading spinner, focus should be set to header
- Set title of the page to something that calls out the type of appointment and the date of the appointment

**Comment by jenniemc** [comment]
@laurenernest @outerpress , the AC of this ticket is modify the document title to include the type of appointment and the date of the appointment for the details page. There is consistency across most detail pages such that the page heading (h1) matches the document title as in `upcoming VA appt`, `upcoming Community Care appt` and `Express Care request`. 

The ones that's inconsistent are `VA appt request` and `CC appt request` (last screenshot). 

QUESTION: keep `VA appt request` and `CC appt request` as defined in AC or change it so that document title matches the h1? 

<hr />
**upcoming VA appointment**

![image](https://user-images.githubusercontent.com/54327023/110863338-8992d380-828e-11eb-8046-d1669818307e.png)
<hr />

**upcoming Community Care appt**
![image](https://user-images.githubusercontent.com/54327023/110863039-15583000-828e-11eb-8557-ad7feff63611.png)
<hr />

**Express Care Request:**
![image](https://user-images.githubusercontent.com/54327023/110863162-40428400-828e-11eb-9f23-6c78002f94e9.png)
<hr />

**VA appt request**
![image](https://user-images.githubusercontent.com/54327023/110864426-1c803d80-8290-11eb-95f3-8b11ea11b2b4.png)

**Comment by laurenernest** [comment]
> QUESTION: keep VA appt request and CC appt request as defined in AC or change it so that document title matches the h1?


A bit of both. [VA/CC] and [Request/Pending] are important context for the contents of the page. There could be 1-3 possible dates of the appointment so that could be misleading if we pick just one for the title. Since both the VA and CC request H1s are `Pending [type of care] appointment`, some combination of [VA/CC] + H1 seems appropriate.
